src/librustc_trans/save/span_utils.rs

```diff
@@ -388,4 +377,40 @@ impl<'a> SpanUtils<'a> {
             self.sess.bug("span errors reached 1000, giving up");
         }
     }
+
+    /// Return true if the span is generated code, and
+    /// it is not a subspan of the root callsite.
+    ///
+    /// Used to filter out spans of minimal value,
+    /// such as references to macro internal variables.
+    pub fn filter_generated(&self, sub_span: Option<Span>, parent: Span) -> bool {
+        if !generated_code(parent) {
+            if sub_span.is_none() {
+                // Non generated code is guaranteed to be Some - this is malformed.
```

Comment:
As in, you should never get here? If so, you should use `unreachable!` or an `assert!`